text/0000-system-receptionist.md

```diff
@@ -0,0 +1,255 @@
+const receptionist = {
+  onRegister(key, actorRef) {
+    // Register actorRef
+    // ...
+
+    observers[key].forEach(observer => {
+      observer.next(registry.get(key))
+    });
+  },
+
+  listener(key) {
+    // Return ActorRef that listens for registry updates on that key
+    // ...
+  }
+}
+```
+
+
+**Unregistering**
+
+At any time, an actor may unregister itself from the system-wide registry. This should notify listener `ActorRef`s that the list of actors under that key has changed.
+
+```ts
+const todoMachine = createMachine({
+  // ...
+  on: {
+    delete: {
+      actions: unregister('todo')
+    }
+  }
+});
+```
+
+This should be roughly equivalent to:
+```ts
+// PSEUDOCODE
+// unregister('todo')
+exit: (context, event, { system, self }) => {
+  system.unregister('todo', self);
+}
+```
+
+## How we teach this
+
+**New concepts:**
+- Actor system
+- System registry
+- Receptionist
+- `onEmit` for invocations (TODO: describe)
+
+**Existing concepts:**
+- Actor refs
+- Actions
+- Subscriptions
+
+No guides will need to be re-organized nor altered. This should be taught after actors (spawning/invoking). There would be two new sections to teach:
+
+- Actor systems (conceptually just a root-level machine that provides shared resources to actors)
+- Actor discovery
+
+This is _not_ required knowledge.
+
+**Teaching:**
+
+- Blog post (discovering actors with receptionists)
+- Actor discovery guide
+
+
+## Use cases
+
+These examples may be contrived, but they should each serve a purpose to highlight challenges that the receptionist feature directly solves.
+
+- An orders actor is registered, and another actor subscribes to the `ORDERS_UPDATED` event. Rather than sending the giant payload of all orders through with the event, a smaller payload such as just the changed item could be sent. If an actor needs more data, `system.receptionist.find('orders').getSnapshot().context.list` could be used. 
+- Common info like auth or user profile data is often required across multiple machines, so referring to the auth/user actor context directly will be a direct line to the data. This should be more efficient than storing it all over the place and would keep queries up to date.
+
+## Drawbacks
+
+This introduces two "new" concepts to XState - systems and receptionists - but it solves many common pain-points in managing and communicating with actors in a straightforward way. Both of these concepts are not new to the actor model, but the additional concepts can feel burdensome to devs who are trying to learn all of XState.
+
+To mitigate this, we should inform devs that XState can be used successfully without systems (orphaned actors by default) nor receptionists, and that the purpose of these two concepts is to _simplify_ common use-cases that are otherwise difficult and complicated to implement.
+
+## Alternatives
+
+TODO
+
+## Unresolved questions
+
+- System options: `createSystem(machine, options?)` (what should go here?)
```

Comment:
it would be even fine if we wouldn't have any options for it - we just need to keep the second argument slot free for potential options in the future